store/postgres/src/store.rs

```diff
@@ -232,15 +330,31 @@ impl BasicStore for Store {
 }
 
 impl StoreTrait for Store {
-    fn event_stream(&mut self) -> Result<Receiver<StoreEvent>, StreamError> {
-        // If possible, create a new channel for streaming store events
-        match self.event_sink {
-            Some(_) => Err(StreamError::AlreadyCreated),
-            None => {
-                let (sink, stream) = channel(100);
-                self.event_sink = Some(sink);
-                Ok(stream)
-            }
+    fn subscribe(
+        &mut self,
+        subgraph: String,
+        entities: Vec<String>,
+    ) -> Box<Stream<Item = EntityChange, Error = ()> + Send> {
+        // Prepare the new subscription by creating a channel and a subscription object
+        let (sender, receiver) = channel(100);
+        let id = Uuid::new_v4().to_string();
+        let subscription = Subscription {
+            subgraph,
+            entities,
+            sender,
+        };
+
+        // Add the new subscription (log an error if somehow the UUID is not unique)
+        let subscriptions = self.subscriptions.clone();
+        let mut subscriptions = subscriptions.write().unwrap();
+        if subscriptions.contains_key(&id) {
+            error!(self.logger, "Duplicate Store subscription ID generated; \
+                                 subscriptions will not work as expected";
+                   "id" => &id);
```

Comment:
This is virtually impossible to happen, we can just remove this check.

Reply:
I'm not sure I want to bet on that (see https://softwareengineering.stackexchange.com/a/130298), even if it's unlikely to ever happen.

Reply:
If we think this can happen then panicking would be more correct than logging and leaving the state inconsistent. Or looping until we generate a different one, unless we're concerned about generating duplicate UUIDs forever.

Reply:
Hey, the looping idea is good. An infinite sequence of colliding UUIDs is _VERY_ unlikely. 😁